include/onnxruntime/core/graph/graph.h

```diff
@@ -167,6 +167,16 @@ class Node {
     return definitions_.output_defs;
   }
 
+  /** Whether current node needs fence check or not. Fence check is needed when either one input or output
+  data needs to be copied cross devices asynchronously, etc CPU<->GPU **/
+  bool NeedFenceCheck() const noexcept {
+    return need_fence_check_;
+  }
+
+  void SetFenceCheck(bool needCheck = false) noexcept {
```

Comment:
nits: 
  - Would be nicer to match the getter name and call this SetNeedFenceCheck. 
  - Also don't think you need to default it to false as that's somewhat counter-intuitive given the method name. 
  - needCheck -> need_check